jcl/src/java.base/share/classes/openj9/management/internal/ThreadInfoBase.java

```diff
@@ -0,0 +1,398 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package openj9.management.internal;
+
+import java.lang.Thread.State;
+import java.util.Arrays;
+import java.util.Objects;
+
+import com.ibm.oti.util.Util;
+
+/**
+ * Container for thread information for use by classes which have access only to the
+ * java.base module.
+ * This is typically wrapped in a ThreadInfo object.
+ */
+public class ThreadInfoBase {
+
+	private final long threadId;
+	private final long nativeTId;
+	private final String threadName;
+	private final Thread.State threadState;
+	private final boolean suspended;
+	private final boolean inNative;
+	private final long blockedCount;
+	private final long blockedTime;
+	private final long waitedCount;
+	private final long waitedTime;
+	private final String lockName;
+	private final long lockOwnerId;
+	private final String lockOwnerName;
+	private StackTraceElement[] stackTrace = new StackTraceElement[0];
+	private final LockInfoBase blockingLockInfo;
+	private LockInfoBase[] lockedSynchronizers = new LockInfoBase[0];
+	private MonitorInfoBase[] lockedMonitors = new MonitorInfoBase[0];
+	private String TOSTRING_VALUE;
```

Comment:
This is a name that usually signals a static constant.
Also, I question the appropriateness of caching the arbitrarily large string (think many stack frames).

Reply:
That is legacy code from `ThreadInfo.java`, including the caching.  I agree that the name was inappropriate and have changed it.

> Also, I question the appropriateness of caching the arbitrarily large string (think many stack frames).

As this change is intended to preserve existing behaviour as much as possible, I suggest opening a separate issue to look at the value of caching.